plugins/poll/lib/poll.rb

```diff
@@ -31,43 +45,103 @@ def self.vote(user, post_id, poll_name, options)
               obj << option.id if option.poll_votes.where(user_id: user.id).exists?
             end
 
-        # remove non-selected votes
-        PollVote.where(poll: poll, user: user).where.not(poll_option_id: new_option_ids).delete_all
+        if poll.irv?
+          # for IRV, we need to remove all votes and re-create them as there is no way to update them due to lack of primary key.
```

Comment:
This is somewhat scary. Maybe it's time we change the db schema to improve the way we store user votes? 🤔

Reply:
I wonder if we should get rid of the `digest` and use "proper" modeling for the options and votes 🤔

Reply:
your call, but this is significant scope creep? :)

Reply:
It is indeed 👍

Reply:
let's keep this in mind but consider for future PR unless significant performance issues are identified.

Reply:
closing for now.